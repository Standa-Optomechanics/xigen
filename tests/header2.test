#!/bin/sh
. $srcdir/defs

cat < $file_expected > $file_ok

expect_out_internal_header
expect_consistent
