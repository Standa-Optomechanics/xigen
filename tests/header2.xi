protocol "v1"
defaults with crc, answer, public

command "reset" writer "rest" (4)
without answer, lock

command "updf" writer "updf" (4)
without public, lock

command "service_command_write_key_impl" writer "wkey" (18) reader "wkey" (15)
with inline
without public
fields:
	int32u key
	reserved 8
answer:
	int8u serviceresult sresult
	reserved 8

command "service_command_goto_firmware_impl" writer "gofw" (4) reader "gofw" (15)
with inline
without public, crc
answer:
	int8u serviceanswer sresult
	reserved 8

command "service_command_has_firmware_impl" writer "hasf" (4) reader "hasf" (15)
with inline
without public, crc
answer:
	int8u serviceanswer sresult
	reserved 8

command "service_command_connect" writer "conn" (14) reader "conn" (15)
without public
with inline
fields:
	reserved 8
answer:
	int8u serviceresult sresult
	reserved 8

command "service_command_disconnect" writer "disc" (14) reader "disc" (15)
without public
with inline
fields:
	reserved 8
answer:
	int8u serviceresult sresult
	reserved 8

command "service_command_write_data" writer "wdat" (143) reader "wdat" (15)
without public
with inline
fields:
	int8u datalen
	int8u data[@datalen]
	reserved 8
answer:
	int8u serviceresult sresult
	reserved 8

command "service_command_write_data1" writer "wdax" (142) reader "wdax" (4)
without public
with inline
fields:
	int8u data[128]
	reserved 8

command "service_command_new_section" writer "nsec" (22) reader "nsec" (15)
without public
with inline
fields:
	int8u data[8]
	reserved 8
answer:
	int8u serviceresult sresult
	reserved 8

command "service_command_metacomm" writer "msec" (22) reader "msec" (15)
without public
with inline
fields:
	int8u data[8] metalen
	reserved 8
answer:
	int8u serviceresult sresult
	reserved 8

/** vim: set ft=c: */
