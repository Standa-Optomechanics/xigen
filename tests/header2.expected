
	#define XIMC_METALEN_SERVICE_COMMAND_METACOMM_DATA 8

	result_t command_updf (device_t id);

	result_t service_command_write_key_impl (device_t id, unsigned int key);

	result_t service_command_goto_firmware_impl (device_t id, service_result_t*sresult);

	result_t service_command_has_firmware_impl (device_t id, service_result_t*sresult);

	result_t service_command_connect (device_t id);

	result_t service_command_disconnect (device_t id);

	result_t service_command_write_data (device_t id, unsigned int datalen, const uint8_t* data);

	result_t service_command_write_data1 (device_t id, const uint8_t* data);

	result_t service_command_new_section (device_t id, const uint8_t* data);

	result_t service_command_metacomm (device_t id, const uint8_t* data);

